docs/references/api/resource_embedding.rst

```diff
@@ -816,6 +816,23 @@ Response:
     }
    }
 
+.. note::
+
+  Selecting a related object right after doing an insert may give unexpected results if said object was modified in a trigger.
+  For example, let's say there's an ``AFTER INSERT`` trigger on ``films`` that inserts a new record to ``technical_specs``.
+  Doing the same ``POST`` request as above, but selecting ``select=title,technical_spec(*)`` instead, it would return:
+
+  .. code-block:: json
+
+     {
+      "title": "127 hours",
+      "techinal_specs": null
+     }
+
+  Since PostgREST does the insertion in a single transaction, the selection is done before the changes are made to the related table, that's why the ``technical_specs`` is empty.
+  In the future we'll implement the insertions of tables and embedded resources at the same time.
+  For now, an alternative is to use :ref:`table_functions` instead.
```

Comment:
Wanted to give an idea of "working on it", but  yeah, I see that it may be too optimistic.

Reply:
I didn't considered this as a bug at first because of our design choice in executing the entire mutation/selection in a single query (I was wrong there, I said "single transaction" instead of "single query"). It's a PostgreSQL expected behavior so the mention of a heads-up seemed OK to me, in case someone encountered this.

But I get how this could be considered a bug if the design is prone to change. I'll open an issue regardless, with a full example to see if it's worth documenting or just track as bug.

Reply:
Added the issue #3286

Also, NVM the workaround, a more complicated query (with json aggregates) needs to be done in the function.